circleguard/gui.py

```diff
@@ -279,71 +307,421 @@ def __init__(self):
+        return all_filled
+
+class MapW(LoadableW):
+    def __init__(self):
+
+        self.map_id_input = InputWidget("Map id", "", "id")
+        self.span_input = InputWidget("Span", "", "normal")
+        self.span_input.field.setPlaceholderText("1-50")
+        self.mods_input = InputWidget("Mods (opt.)", "", "normal")
+
+        super().__init__("Map", [self.map_id_input, self.span_input])
+
+        self.layout.addWidget(self.map_id_input, 1, 0, 1, 8)
+        self.layout.addWidget(self.span_input, 2, 0, 1, 8)
+        self.layout.addWidget(self.mods_input, 3, 0, 1, 8)
+
+    def check_required_fields(self):
+        for input_widget in self.required_input_widgets:
+            all_filled = True
+            # don't count span_input as empty when it has placeholder text
+            filled = input_widget.field.text() != "" or input_widget.field.placeholderText() != ""
+            if not filled:
+                input_widget.show_required()
+                all_filled = False
+        return all_filled
+
+class UserW(LoadableW):
+    def __init__(self):
+
+        self.user_id_input = InputWidget("User id", "", "id")
+        self.span_input = InputWidget("Span", "", "normal")
+        self.mods_input = InputWidget("Mods (opt.)", "", "normal")
+
+        super().__init__("User", [self.user_id_input, self.span_input])
+
+        self.layout.addWidget(self.user_id_input, 1, 0, 1, 8)
+        self.layout.addWidget(self.span_input, 2, 0, 1, 8)
+        self.layout.addWidget(self.mods_input, 3, 0, 1, 8)
+
+
+class MapUserW(LoadableW):
+    def __init__(self):
+        self.map_id_input = InputWidget("Map id", "", "id")
+        self.user_id_input = InputWidget("User id", "", "id")
+        self.span_input = InputWidget("Span", "", "normal")
+
+        super().__init__("Map User", [self.map_id_input, self.user_id_input, self.span_input])
+
+        self.layout.addWidget(self.map_id_input, 1, 0, 1, 8)
+        self.layout.addWidget(self.user_id_input, 2, 0, 1, 8)
+        self.layout.addWidget(self.span_input, 3, 0, 1, 8)
+
+
+class MainTab(QFrame):
+    set_progressbar_signal = pyqtSignal(int) # max progress
+    increment_progressbar_signal = pyqtSignal(int) # increment value
     update_label_signal = pyqtSignal(str)
     write_to_terminal_signal = pyqtSignal(str)
-    add_comparison_result_signal = pyqtSignal(object)  # Result
+    add_result_signal = pyqtSignal(object) # Result
     add_run_to_queue_signal = pyqtSignal(object) # Run object (or a subclass)
     update_run_status_signal = pyqtSignal(int, str) # run_id, status_str
+    print_results_signal = pyqtSignal() # called after a run finishes to flush the results queue before printing "Done"
 
-    TAB_REGISTER = [
-        {"name": "MAP"},
-        {"name": "SCREEN"},
-        {"name": "LOCAL"},
-        {"name": "VERIFY"},
-        {"name": "VISUALIZE"}
-    ]
+    LOADABLES_COMBOBOX_REGISTRY = ["Replay Map", "Replay Path", "Map", "User", "Map User"]
```

Comment:
```suggestion
    LOADABLES_COMBOBOX_REGISTRY = ["User Replay on Map", "Local Replay", "Map", "User", "All Users Replays on Map"]
```
Last one is a bit meh but couldn't think of a better name